framework/src/modules/interoperability/mainchain/commands/cc_update.ts

```diff
@@ -47,114 +43,48 @@ import {
 import { MainchainInteroperabilityInternalMethod } from '../store';
 
 export class MainchainCCUpdateCommand extends BaseCrossChainUpdateCommand {
 	public async verify(
 		context: CommandVerifyContext<CrossChainUpdateTransactionParams>,
 	): Promise<VerificationResult> {
-		const { params: txParams } = context;
-
-		try {
-			validator.validate(crossChainUpdateTransactionParams, context.params);
-		} catch (err) {
-			return {
-				status: VerifyStatus.FAIL,
-				error: err as Error,
-			};
-		}
-
-		const partnerChainStore = this.stores.get(ChainAccountStore);
-		const partnerChainAccount = await partnerChainStore.get(context, txParams.sendingChainID);
-
-		// Section: Liveness of Partner Chain
-		if (partnerChainAccount.status === ChainStatus.TERMINATED) {
-			return {
-				status: VerifyStatus.FAIL,
-				error: new Error(
-					`Sending partner chain ${txParams.sendingChainID.readInt32BE(0)} is terminated.`,
-				),
-			};
-		}
-		const interoperabilityInternalMethod = this.getInteroperabilityInternalMethod(context);
-		if (partnerChainAccount.status === ChainStatus.ACTIVE) {
-			const isLive = await interoperabilityInternalMethod.isLive(
-				txParams.sendingChainID,
-				Date.now(),
-			);
-			if (!isLive) {
-				return {
-					status: VerifyStatus.FAIL,
-					error: new Error(
-						`Sending partner chain ${txParams.sendingChainID.readInt32BE(0)} is not live.`,
-					),
-				};
-			}
-		}
-
-		// Section: Liveness Requirement for the First CCU
-		const livenessRequirementFirstCCU = checkLivenessRequirementFirstCCU(
-			partnerChainAccount,
-			txParams,
+		const { params } = context;
+		validator.validate<CrossChainUpdateTransactionParams>(
+			crossChainUpdateTransactionParams,
+			context.params,
 		);
-		if (livenessRequirementFirstCCU.error) {
-			return livenessRequirementFirstCCU;
-		}
-		// Section: Certificate and Validators Update Validity
-		const certificateValidity = checkCertificateValidity(partnerChainAccount, txParams.certificate);
+		const internalMethod = this.getInteroperabilityInternalMethod(context);
 
-		if (certificateValidity.error) {
-			return certificateValidity;
+		// TODO: Update isLive
+		const isLive = await internalMethod.isLive(params.sendingChainID, context.header.timestamp);
+		if (!isLive) {
+			throw new Error('The sending chain is not live.');
 		}
 
-		const partnerValidatorStore = this.stores.get(ChainValidatorsStore);
-		const partnerValidators = await partnerValidatorStore.get(context, txParams.sendingChainID);
-		// If params contains a non-empty activeValidatorsUpdate and non-empty certificate
-		const validatorsHashValidity = checkValidatorsHashWithCertificate(txParams, partnerValidators);
-		if (validatorsHashValidity.error) {
-			return validatorsHashValidity;
+		const sendingChainAccount = await this.stores
+			.get(ChainAccountStore)
+			.get(context, params.sendingChainID);
+		if (sendingChainAccount.status === ChainStatus.REGISTERED && params.certificate.length === 0) {
```

Comment:
I think we are missing liveness condition for first CCU with a non-empty inbox update,
```python
if chainAccount(ccu.params.sendingChainID).status == CHAIN_STATUS_REGISTERED:
        verifyLivenessConditionForRegisteredChains(ccu)
```